x/bsonx/bsoncore/document.go

```diff
@@ -291,6 +291,91 @@ func (d Document) String() string {
 	return buf.String()
 }
 
+// StringN stringifies a document upto N bytes
+func (d Document) StringN(n int) string {
```

Comment:
There's a lot of code here that's almost identical to the `String` method. We should find a way to prevent this duplication. One option is to call `StringN` from `String` and pass something like `math.MaxInt` that would prevent ever actually truncating.

This suggestion applies to the `Array`, `Element`, and `Value` types as well.